be/src/runtime/stream_load/stream_load_executor.cpp

```diff
@@ -230,6 +231,39 @@ Status StreamLoadExecutor::commit_txn(StreamLoadContext* ctx) {
         LOG(WARNING) << "commit transaction failed, errmsg=" << status.get_error_msg() << ctx->brief();
         if (status.code() == TStatusCode::PUBLISH_TIMEOUT) {
             ctx->need_rollback = false;
+            if (config::enable_sync_publish) {
+                //wait for apply finish
+                TGetLoadTxnStatusRequest v_request;
+                TGetLoadTxnStatusResult v_result;
+                set_request_auth(&v_request, ctx->auth);
+                v_request.db = ctx->db;
+                v_request.tbl = ctx->table;
+                v_request.txnId = ctx->txn_id;
+                int64_t load_deadline_sec =
+                        std::min(ctx->load_deadline_sec, UnixSeconds() + config::txn_wait_publish_timeout_sec);
+                while (load_deadline_sec > UnixSeconds() + 60 + config::txn_commit_rpc_timeout_ms / 1000) {
+                    sleep(60);
```

Comment:
60s may be too large for retry.

Reply:
> 60s may be too large for retry. It could hang http worker thread for a long time.

Actually we hang the http worker during the whole stream load task. We send commit transaction rpc to FE and wait publish finish in FE. If publish success, we will return client success. If publish timeout, we return FE visible for a while.  And now we will hang the http worker until apply finish because i think apply is a part of publish. If we return in advance, we may return a lot of `visible for a while` in stream load task which i think is not acceptable.